dev/DynamicDependency/MddBootstrap.cpp

```diff
@@ -0,0 +1,378 @@
+        if ((id.size() >= maxIdLength) ||
+            (swscanf_s(id.c_str(), L"ddlm-%hu.%hu.%hu.%hu-%9s", &version.Major, &version.Minor, &version.Build, &version.Revision, architectureAsString, static_cast<unsigned>(ARRAYSIZE(architectureAsString))) != 5))
+        {
+            (void)LOG_HR_MSG(ERROR_INVALID_DATA, "%ls", id.c_str());
+            continue;
+        }
+
+        // Does the version meet the minVersion criteria?
+        if (version.Version < minVersion.Version)
+        {
+            continue;
+        }
+
+        // Does the architecture match?
+        const auto architecture{ ParseArchitecture(architectureAsString) };
+        if (architecture != GetCurrentArchitecture())
+        {
+            continue;
+        }
+
+        // Do we have a package under consideration?
+        if (!foundAny)
+        {
+            bestFitVersion = version;
+            bestFitClsid = GetClsid(appExtension);
+            foundAny = true;
+            continue;
+        }
+
+        // Do we already have a higher version under consideration?
+        if (bestFitVersion.Version < version.Version)
+        {
+            bestFitVersion = version;
+            bestFitClsid = GetClsid(appExtension);
+            continue;
+        }
+    }
+    THROW_HR_IF_MSG(HRESULT_FROM_WIN32(ERROR_NO_MATCH), !foundAny, "AppExtension.Name=%ls, MinVersion=%hu.%hu.%hu.%hu",
+                    appExtensionName, minVersion.Major, minVersion.Minor, minVersion.Build, minVersion.Revision);
+    return bestFitClsid;
+}
+
+CLSID GetClsid(const winrt::Windows::ApplicationModel::AppExtensions::AppExtension appExtension)
+{
+    const auto properties{ appExtension.GetExtensionPropertiesAsync().get() };
+    auto propertiesClsid{ properties.Lookup(L"CLSID").as<winrt::Windows::Foundation::Collections::IPropertySet>() };
+    auto value{ propertiesClsid.Lookup(L"#text").as<winrt::Windows::Foundation::IPropertyValue>() };
+    THROW_HR_IF_NULL(E_UNEXPECTED, value);
+    THROW_HR_IF(E_UNEXPECTED, value.Type() != winrt::Windows::Foundation::PropertyType::String);
+    const auto text{ value.GetString() };
+
+    // Convert the CLSID as a string to a CLSID as a GUID
+    // Problem: CLSIDFromString() also does a lookup for a registered object by the CLSID.
+    //          We just want the string->GUID conversion, not any additional work.
+    // Workaround this by using UuidFromString()
+
+    // Problem: UuidFromString() takes a RPC_WSTR but that's defined as unsigned short*
+    //          unless RPC_USE_NATIVE_WCHAR is defined.
+    // Workaround this with casts. Include some asserts to verify we're not misusing memory.
+
+    auto textString{ const_cast<PWSTR>(text.c_str()) };
+    auto textRpcString{ reinterpret_cast<RPC_WSTR>(textString) };
+
+    static_assert(sizeof(textString) == sizeof(textRpcString));
+    static_assert(sizeof(textString[0]) == sizeof(textRpcString[0]));
+
+    UUID clsid{};
+    THROW_IF_WIN32_ERROR(UuidFromStringW(textRpcString, &clsid));
```

Comment:
`winrt::guid` can parse a stringified guid now! https://github.com/microsoft/cppwinrt/pull/833/files

Reply:
Not sure we're using the latest c++/winrt and Scott's updating that so I'd rather not have 'merge fun' across vcxproj

Added to the todo list for follow up